### DISCUSSION TITLE
sidekiq.inline! - enqueued jobs

### DISCUSSION THREAD

**Discussion by byellokore** [sidekiq/sidekiq#5582] [Q&A]
Hi, the docs here [](https://github.com/mperham/sidekiq/wiki/Testing) are crispy clear... But I have a quick question.

Sidekiq.inline! will actually enqueue the jobs in redis or it will just enqueue in the array as sidekiq.fake! and execute them directly from the jobs array?  or it just runs the job right away?

**Comment by mperham** [comment]
I would suggest you dive into the code and find out for yourself. You'll probably learn a ton about Sidekiq in doing so. `bundle open sidekiq`

**Reply by byellokore** [reply]
I'm trying @mperham . We use Sidekiq enterprise. I wrote a test to for Job that calls another job and I was asked to replace the inline! with just `Job.new.perform`. I'm wondering if it we should never use the inline then? Anyway, I will dive in the code. Thank you.

**Reply by byellokore** [reply]
And wiki has

<img width="608" alt="Capture d’écran, le 2022-10-21 à 11 58 44" src="https://user-images.githubusercontent.com/10491484/197239025-a71d7564-d4a2-4094-9caf-0cd74e97efbf.png">

and later

<img width="934" alt="Capture d’écran, le 2022-10-21 à 11 58 16" src="https://user-images.githubusercontent.com/10491484/197239076-8c10b41d-56c2-4c01-8abf-53b9bdd01e42.png">

maybe it's just my level of english but I got confused with this.

**Reply by mperham** [reply]
Thanks, I fixed the wiki content.

**Reply by byellokore** [reply]
Thanks! @mperham I appreciate your time.
